pkg/prober/prober.go

```diff
@@ -0,0 +1,157 @@
+package prober
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/component"
+	"github.com/prometheus/common/route"
+)
+
+const (
+	healthyEndpointPath  = "/-/healthy"
+	readyEndpointPath    = "/-/ready"
+	okProbeText          = "thanos %v is %v"
+	errorProbeText       = "thanos %v is not %v. Reason: %v"
+	probeErrorHTTPStatus = 503
+	initialErrorText     = "thanos %s is initializing"
+)
+
+// Prober represents health and readriness status of given compoent.
+type Prober struct {
+	logger       log.Logger
+	componentMtx sync.Mutex
+	component    component.Component
+	readyMtx     sync.Mutex
+	readiness    error
+	healthyMtx   sync.Mutex
+	healthiness  error
+}
+
+// SetLogger sets logger used by the Prober.
+func (p *Prober) SetLogger(logger log.Logger) {
+	p.logger = logger
+}
+
+func (p *Prober) getLogger() log.Logger {
+	return p.logger
+}
+
+// SetComponent sets component name of the Prober displayed in responses.
+func (p *Prober) SetComponent(component component.Component) {
+	p.componentMtx.Lock()
+	defer p.componentMtx.Unlock()
+	p.component = component
+}
+
+func (p *Prober) getComponent() component.Component {
+	p.componentMtx.Lock()
+	defer p.componentMtx.Unlock()
+	return p.component
+}
+
+// NewProber returns Prober reprezenting readi	ness and healthiness of given component.
+func NewProber(component component.Component, logger log.Logger) *Prober {
+	initialErr := fmt.Errorf(initialErrorText, component)
+	prober := &Prober{}
+	prober.SetComponent(component)
+	prober.SetLogger(logger)
+	prober.SetNotHealthy(initialErr)
+	prober.SetNotReady(initialErr)
+	return prober
+}
+
+// HandleInMux registers readiness and liveness probes to mux.
```

Comment:
This comment seems off. The method is called `RegisterInRouter`.

Reply:
Thanks, it was leftover after refactoring.